test/wallet-http.js

```diff
@@ -0,0 +1,650 @@
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const json = await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000,
+      debug: true
+    });
+
+    const bids = json.outputs.filter(output => output.covenant.type === types.BID);
+    assert.equal(bids.length, 1);
+
+    const [bid] = bids;
+    assert.equal(bid.covenant.items.length, 4);
+
+    const [nameHash, start, rawName, blind] = bid.covenant.items;
+    assert.equal(nameHash, rules.hashName(name).toString('hex'));
+
+    // initially opened in the first block mined, so chain.height + 1
+    const hex = Buffer.from(start, 'hex').reverse().toString('hex');
+    assert.equal(parseInt(hex, 16), info.chain.height + 1);
+
+    assert.equal(rawName, Buffer.from(name, 'ascii').toString('hex'));
+
+    // blind is type string, so 32 * 2
+    assert.equal(blind.length, 32 * 2);
+  });
+
+  it('should get name info', async () => {
+    const names = await wallet.getNames();
+
+    assert(names.length > 0);
+    const [ns] = names;
+
+    const nameInfo = await wallet.getName(ns.name);
+
+    assert.deepEqual(ns, nameInfo);
+  });
+
+  it('should fail to open a bid without a bid value', async () => {
+    const fn = async () => (await wallet.createBid({
+      name: name
+    }));
+
+    assert.rejects(fn);
+  });
+
+  it('should fail to open a bid without a lockup value', async () => {
+    const fn = async () => (await wallet.createBid({
+      name: name,
+      bid: 1000
+    }));
+
+    assert.rejects(fn);
+  });
+
+  it('should create a reveal', async () => {
+    await wallet.createOpen({
+      name: name
+    });
+
+    const {treeInterval} = network.names;
+    for (let i = 0; i < treeInterval + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    await wallet.createBid({
+      name: name,
+      bid: 1000,
+      lockup: 2000
+    });
+
+    const {biddingPeriod} = network.names;
+    for (let i = 0; i < biddingPeriod + 1; i++)
+      await nclient.execute('generatetoaddress', [1, cbAddress]);
+
+    await sleep(100);
+
+    const {info} = await nclient.execute('getnameinfo', [name]);
+    assert.equal(info.name, name);
+    assert.equal(info.state, 'REVEAL');
+
+    const json = await wallet.createReveal({
+      name: name
+    });
+
+    const reveals = json.outputs.filter(output => output.covenant.type === types.REVEAL);
+    assert.equal(reveals.length, 1);
+  });
+
+  it('should get auction info', async () => {
+    const names = await wallet.getNames();
+
+    await sleep(100);
+
+    assert(names.length > 0);
+    const [,ns] = names;
+
+    const auction = await wallet.getAuction(ns.name);
+
+    // auction info returns a list of bids
+    // and a list of reveals for the name
+    assert.ok(auction.bids);
+    assert.ok(auction.reveals);
```

Comment:
I think `assert.ok([])` passes, might want to tighten this test.

Reply:
Actually when I `console.log(auction)` right here I get `reveals: []` -- shouldn't there be a `REVEAL` from the last test?

Reply:
@tynes was this addressed. I see it is associated with an outdated commit, but the issue has not been resolved.

Reply:
@boymanjor - To test this properly, I think it needs a full node started from scratch so that its not too tightly coupled to previous tests. It currently really only tests that `bids` and `reveals` are properties on the object returned.

Reply:
@pinheadmz - tightened up testing here, using state from a previous test and making clear assertions on what the state should be. 👍